lightningd/plugin.c

```diff
@@ -1442,22 +1495,30 @@ static const char *plugin_rpcmethod_add(struct plugin *plugin,
 	else
 		return tal_fmt(plugin,
 			    "\"usage\" not provided by plugin");
-
+	plugin_rpcmethod_clnrest_add(plugin, buffer, meth, cmd);
 	err = json_parse_deprecated(cmd, buffer, deprtok, &cmd->depr_start, &cmd->depr_end);
 	if (err)
 		return tal_steal(plugin, err);
 
 	cmd->dev_only = false;
 	cmd->dispatch = plugin_rpcmethod_dispatch;
 	cmd->check = plugin_rpcmethod_check;
-	if (!jsonrpc_command_add(plugin->plugins->ld->jsonrpc, cmd, usage)) {
-		struct plugin *p =
-		    find_plugin_for_command(plugin->plugins->ld, cmd->name);
+	char *collision_name;
+	if (!jsonrpc_command_add(plugin->plugins->ld->jsonrpc, cmd, usage,
+				 &collision_name)) {
+		struct plugin *p = find_plugin_for_command(plugin->plugins->ld,
+							   collision_name);
+		if (!p) {
```

Comment:
maybe add a note about why a plugin wouldn't be found? (eg. two methods on the same plugin with the same path?)